coresdk/src/coresdk/database.h

```diff
@@ -1,60 +1,301 @@
-//
-//  database.h
-//  splashkit
-//
-//  Created by Jake Renzella on 2/08/2016.
-//  Copyright © 2016 Andrew Cain. All rights reserved.
-//
+/**
+ * @header Database
+ * @author Jake Renzella
+ * @brief SplashKit Database allows you to create, query and otherwise manipulate a database.
+ *
+ * The SplashKit's database library allows you to easily create or load databases
+ * and to perform queries on them. To get started with databases in SplashKit,
+ * the first thinkg you need to do is to load or create a database file. You can
+ * do this by calling the `open_database(string name)` function.
+ */
 
 #ifndef database_h
 #define database_h
 
 #include <string>
 using namespace std;
 
+/**
+ * The `database` type is used to refer to databases that can be
+ * manipulated by the SplashKit database code. Databases are:
+ *
+ *   - opened with `open_database`,
+ *
+ *   - and must be released using `free_database` (to release a specific
+ *     database) or `delete_all_databases` (to release all loaded
+ *     databases).
+ *
+ * @attribute class database
+ */
 typedef struct _database *database;
+
+/**
+ * The `query_result` type is used to store the result of performing
+ * a query on the database. It can:
+ *
+ *   - show the success or failure of a query using `query_success`,
+ *
+ *   - stores the data from queries which return data. This data is
+ *     accessed using `query_column_for_bool`, `query_column_for_int`,
+ *     `query_column_for_string` and `query_column_for_double`.
+ *
+ * @attribute class query_result
+ */
 typedef struct sk_query_result *query_result;
 
+
+/**
+ * Determines if SplashKit has a database loaded for the supplied name.
+ * This checks against all databases loaded, those loaded without a name
+ * are assigned the filename as a default. If this returns `false`, you may
+ * want to use `load_database` to load in a specific database and give
+ * it the desired name.
+ *
+ * @param name The name to check if a database is loaded.
+ *
+ * @returns Returns `true` if a database with the given `name` has
+ * has been loaded.
+ *
+ * @attribute class   database
+ * @attribute method  has_database
+ */
 bool has_database(string name);
 
+/**
+ * Retrieves a `database` that has been loaded into SplashKit.
+ *
+ * @param name The name of the database to return.
+ *
+ * @returns Returns the `database` that has been loaded with the specified
+ *          `name` using `open_database`.
+ *
+ * @attribute class   database
+ * @attribute method  database_named
+ */
 database database_named(string name);
 
+/**
+ * Calculates the number of row changed in the database at the last time
+ * a changing query was performed on the given database.
+ *
+ * @param db The database to check how many rows changed.
+ *
+ * @returns Returns the `int` which represents how many rows were changed
+ *          in the database.
+ *
+ * @attribute class   database
+ * @attribute method  rows_changed
+ */
 int rows_changed(database db);
```

Comment:
Yeah if we have two overloaded functions, then one can use the `self` attribute (using db) and the other one wouldn't.